src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -3467,11 +3467,11 @@ trait Types
      *  bounds are exactly tp.
      */
     override def bounds: TypeBounds = underlying.bounds match {
-      case TypeBounds(_: this.type, _: this.type) => TypeBounds(this, this)
-      case oftp                                   => oftp
+      case bd @ TypeBounds(_: this.type, _: this.type) => bd
```

Comment:
I've often wondered why this was done, but note that `TypeBounds(this, this)` hashconses (using `unique`), so it could relieve memory pressure.